origins_workflow/origins_workflow.module

```diff
@@ -278,3 +266,106 @@ function _audit_link($field, $dt, $nid) {
+      $variables['header']['select_column_two']['content'] = ['#markup' => t('Compare revision 2')];
+    }
+    // Loop through the rows and add the revision number column to each.
+    foreach ($variables['rows'] as $idx => $row) {
+      $this_version = '';
+      if (isset($row['cells'][3]['content']['#links'])) {
+        // Cheekily extract current revision from the 'revert' button link.
+        $this_url = $row['cells'][3]['content']['#links']['revert']['url'];
+        if (isset($this_url)) {
+          $this_version = $this_url->getRouteParameters()['node_revision'];
+        }
+        // Change the link text from 'Set as current revision' to
+        // 'Copy to new revision' as it makes more sense - see issue
+        // https://www.drupal.org/project/drupal/issues/2899719
+        if (isset($variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'])) {
+          $title_markup = $variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'];
+          // Check the existing text in an attempt to detect any core fix that
+          // might come along.
+          $title_text = $title_markup->getUntranslatedString();
+          if (($title_text == 'Set as current revision') || ($title_text == 'Revert')) {
+            $variables['rows'][$idx]['cells'][3]['content']['#links']['revert']['title'] = t('Copy to new revision');
+          }
+        }
+      }
+      else {
+        // There is no 'revert' button in this row, therefore it must be the
+        // current versionand we need to add a 'Create Draft of Published' link.
+        $this_version = _create_draft_link_for_revision($variables, $idx, $row);
+      }
+      $new_col = ['tag' => 'td', 'content' => ['#markup' => $this_version]];
+      // Insert this new column after the first one.
+      // Use array_splice as we don't know exactly how many columns there
+      // will be (and this code should work whether the 'diff' module has
+      // been enabled or not).
+      array_splice($variables['rows'][$idx]['cells'], 1, 0, [$new_col]);
+    }
+  }
+}
+
+/**
+ * Add a 'Create Draft of Published' link.
+ */
+function _create_draft_link_for_revision(&$variables, $idx, $row) {
+  // Cannot manage to extract the revision id from this row
+  // anywhere so we'll (reluctantly) have to load up the full node.
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $this_version = $node->getRevisionId();
+  // Now change the 'Current Revision' markup to a
+  // 'Create Draft of Published' link.
+  if ($idx > 0) {
+    // Just check that there is markup in there and no link (just in case
+    // core has been updated in the meantime, in which case this fix may
+    // not be needed).
+    if (isset($row['cells'][3]['content']['#markup']) && !isset($row['cells'][3]['content']['#links'])) {
+      $markup = $row['cells'][3]['content']['#markup'];
+      // There is markup in there, just double check that
```

Comment:
This is new code to create a 'Create Draft of Published' link to replace the old 'Current revision' markup text